src/etc/inc/services.inc

```diff
@@ -1274,7 +1274,7 @@ function dhcpdkey($dhcpifconf) {
 		$algorithm = empty($dhcpifconf['ddnsdomainkeyalgorithm']) ? 'hmac-md5' : $dhcpifconf['ddnsdomainkeyalgorithm'];
 		$dhcpdconf .= "key {$dhcpifconf['ddnsdomainkeyname']} {\n";
 		$dhcpdconf .= "	algorithm {$algorithm};\n";
-		$dhcpdconf .= "	secret {$dhcpifconf['ddnsdomainkey']};\n";
+		$dhcpdconf .= "	secret \"{$dhcpifconf['ddnsdomainkey']}\";\n";
```

Comment:
The value inside should have double quotes escaped, or else you end up with another situation like https://redmine.pfsense.org/issues/10295

Reply:
This is incorrect too.
dnssec-keygen should be used, 
dhcpd.conf(5):
>       You should choose your own secret key, of course.  The ISC BIND 9  dis-
>        tribution  comes	 with  a  program  for	generating  secret keys	called
>        dnssec-keygen.  If you are using	BIND 9's dnssec-keygen,	the above  key
>        would be	created	as follows:
> 
> 	    dnssec-keygen -a HMAC-MD5 -b 128 -n	USER DHCP_UPDATER
> 
>        The  key	 name, algorithm, and secret must match	that being used	by the
>        DNS server.

I tried to use hash_hmac(), but it needs testing

Reply:
That would be to generate a new key, but in this case the user would be entering a key generated elsewhere (on the actual name server) so it has to be a text input.

Reply:
all done
not it fixes DynDNS key using strict base64_decode()